build/lib/release.sh

```diff
@@ -97,31 +108,30 @@ function kube::release::package_tarballs() {
 function kube::release::package_src_tarball() {
   local -r src_tarball="${RELEASE_TARS}/kubernetes-src.tar.gz"
   kube::log::status "Building tarball: src"
-  if [[ "${KUBE_GIT_TREE_STATE-}" == "clean" ]]; then
+  if [ "${KUBE_GIT_TREE_STATE-}" = 'clean' ]; then
```

Comment:
what's the reason for [[ => [ ? this file is marked as bash?

Reply:
I'd say the same reason why one, in C, would write

if (a == b)

and not

if ((a == b))

Sure, the shell used is bash but the test content is not bash specific.

> we're fairly consistent about using `[[ ]]` in bash scripts.

Out of the 322 shell scripts (at least those with the file name extension .sh) currently on master, 
312 of them being bash ones, the single bracket `[ ]` expression is found in 282 (88%) of the total 322. So I'd say there is quite some dynamic in the use of "fairly" in that sentence. :-)

Nevertheless, if you do strongly want every single bracket expression to be `[[ ]]` in /build/lib/release.sh just for the sake of it, I can of course do that. No problem, just let me know.

Reply:
`(a == b)` and `((a == b))` are actually the same C code though, `[[ foo ]]` and `[ foo ]` are not the same bash code.

>  ... So I'd say there is quite some dynamic in the use of "fairly" in that sentence. :-)

fair, there's some really old code that had no consistency of any sort. but only a few of us that actively touch it, and we've been more consistently settled on `[[` unless we're trying to write sh.

I have less strong opinions on it myself, other than we also avoid diffs that don't functionally improve things, unless they are enforced by some consistent lint /shrug

Reply:
> `(a == b)` and `((a == b))` are actually the same C code though, `[[ foo ]]` and `[ foo ]` are not the same bash code.

Fair point, in a behind-the-scene sense, but not really from the programmer's view.

> that actively touch it, and we've been more consistently settled on `[[` unless we're trying to write sh.

I interpret that as a not entirely clear way of writing "I won't ever let this pass until the '[[ foo ]]' is back", so I will do an update.

Reply:
not at all, just perhaps hopeful for a smaller diff :+)

we've had a lot of breakage with seemingly innocuous bash changes, smaller diffs mean a quicker turnaround reading everything thoroughly and more confidence approving them.

An awful lot of shell "fixes" people have merged broke things :/

Reply:
(I do think using double brace in bash is best to avoid accidentally using single where it matters, but I'm happy to be taught otherwise :-))

Reply:
Since the update I did yesterday morning means those brace changes are removed from this patchset to meet your wishes, I hope this patchset can now move forward.